peer/gossip.go

```diff
@@ -62,14 +62,19 @@ func (g *Gossiper) Gossip(ctx context.Context, contentID []byte, subnet *id.Hash
 		}
 	}
 
+	var wg sync.WaitGroup
+	wg.Add(len(recipients))
 	msg := wire.Msg{Version: wire.MsgVersion1, To: *subnet, Type: wire.MsgTypePush, Data: contentID}
-	for _, recipient := range recipients {
-		innerContext, cancel := context.WithTimeout(ctx, g.opts.Timeout)
-		if err := g.transport.Send(innerContext, recipient, msg); err != nil {
-			g.opts.Logger.Error("pushing gossip", zap.String("peer", recipient.String()), zap.Error(err))
-		}
-		cancel()
-	}
+	for i := range recipients {
+		go func(index int) {
+			defer wg.Done()
+			recipient := &recipients[index]
+			if err := g.transport.Send(ctx, *recipient, msg); err != nil {
+				g.opts.Logger.DPanic("pushing gossip", zap.String("peer", recipient.String()), zap.Error(err))
```

Comment:
We should probably stick with an `Error` log here as opposed to `DPanic`.

Reply:
Done

Reply:
Done